src/phases/obtain-status-effect-phase.ts

```diff
@@ -38,9 +38,10 @@ export class ObtainStatusEffectPhase extends PokemonPhase {
 
   public override start(): void {
     const pokemon = this.getPokemon();
-    if (pokemon && !pokemon.status) {
+    if (pokemon && !pokemon.hasNonVolatileStatusEffect(false, true)) {
       if (pokemon.trySetStatus(this.statusEffect, false, this.sourcePokemon)) {
         if (this.turnsRemaining) {
+          // @ts-expect-error - `Pokemon#status` is protected; TODO: change this
           pokemon.status!.sleepTurnsRemaining = this.turnsRemaining;
```

Comment:
```suggestion
          /** @todo Change the accessibility of `Pokemon#status` */
          pokemon["status"]!.sleepTurnsRemaining = this.turnsRemaining;
```

Alternatively, you could make a `Pokemon` method to set the Pokemon's `sleepTurnsRemaining`